src/pages/Main.js

```diff
@@ -1,66 +1,189 @@
+                }
+            } else {
+                setData(rawData)
+            }
+        } else if (tag === "hobbies") {
+            if (midTag !== "") {
+                if (midTag === "include") {
+                    const filter2Data = data.filter((profile) => {
+                        return profile[tag] === filterInput;
+                    })
+                    setData(filter2Data)
+                }
+                if (midTag === "exclude") {
+                    const filter2Data = data.filter((profile) => {
+                        return profile[tag] !== filterInput;
+                    })
+                    setData(filter2Data)
+                }
+            }
+        }else if(tag==="date_of_birth"){
+            if(midTag!==""){
+                if(midTag==="is"){
+                    const filter2Data=data.filter((profile)=>{
+                        const incomingDateISO= profile[tag].split('/');
+                        const formattedDate= incomingDateISO[2] + '-' + incomingDateISO[0].padStart(2, '0') + '-' + incomingDateISO[1].padStart(2, '0');
+                        return formattedDate === (filterInput).replace(/\//g, '-');
+                    })
+                    setData(filter2Data)
+                }
+                if(midTag==="before"){
+                    const filter2Data=data.filter((profile)=>{
+                        const incomingDateISO= profile[tag].split('/');
+                        const formattedDate= incomingDateISO[2] + '-' + incomingDateISO[0].padStart(2, '0') + '-' + incomingDateISO[1].padStart(2, '0');
+                        return formattedDate < (filterInput).replace(/\//g, '-');
+                    })
+                    setData(filter2Data)    
+                }
+                if(midTag==="after"){
+                    const filter2Data=data.filter((profile)=>{
+                        const incomingDateISO= profile[tag].split('/');
+                        const formattedDate= incomingDateISO[2] + '-' + incomingDateISO[0].padStart(2, '0') + '-' + incomingDateISO[1].padStart(2, '0');
+                        return formattedDate > (filterInput).replace(/\//g, '-');
+                    })
+                    setData(filter2Data)   
+                }
+            }
+
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [tag, midTag, filterInput])
 
+    console.log(midTag)
     return (
         <div>
-            <h1 className='logo'>Filter</h1>
-            <Filters tag={tag} handleTag={handleTag} data={data} tagsArray={tagsArray} filter={filter}/>
-            <Display tag={tag} data={data} filter={filter} />
+            <a href="index.html" className='logo'>Filter</a>
+            <Filters tag={tag} handleTag={handleTag} data={data} tagsArray={rawData} filter={filter} handleMidTag={handleMidTag} midTag={midTag} handleFilterInput={handleFilterInput} resetData={resetData} filterInput={filterInput} />
```

Comment:
as mentioned, group similar props in objects and pass as props, this makes the code cleaner and more readable